ob1k-core/src/main/java/com/outbrain/ob1k/client/targets/TargetProvider.java

```diff
@@ -1,11 +1,35 @@
 package com.outbrain.ob1k.client.targets;
 
+import java.util.List;
+
 /**
  * An API for retrieving targets for the client invocations.
  * An example target may be a static list of hosts + round robin, or a discovery based provider.
+ *
  * @author eran 6/21/15.
  */
 public interface TargetProvider {
-    String getTargetLogicalName();
-    String provideTarget();
+  /**
+   * Returns logical name of the target provider (e.g. client name).
+   *
+   * @return target logical name
+   */
+  String getTargetLogicalName();
+
+  /**
+   * Returns remote target for invocation.
+   *
+   * @return target host
+   */
+  String provideTarget();
+
+  /**
+   * Returns collection of remote targets for invocation for size specified.
+   * May return collection with duplicate targets, and the collection size will always
+   * be as requested
+   *
+   * @param targetsNum number of targets to return
+   * @return targets hosts
+   */
+  List<String> provideTargets(int targetsNum);
```

Comment:
3 methods are 2 methods too much.
let's move to one.
also we might want to switch to InetAddress instead of String, it's too generic. or URI if we want to support url prefix.
also I agree with Haim, List is too specific. does the order of the list represent anything or is it really a set ?
I also think that the parameter should be called maxTargetSetSize and should not contains duplicates if the number is bigger than the actual size.